### PR TITLE
Add fuzzy search for mentions

### LINKED ISSUE #470: Add fuzzy search for mentions and slash commands
References:
1. https://www.npmjs.com/package/fuzzy-search
2. https://fusejs.io/

- Permit spaces in the mention suggestions.

### PR DESCRIPTION
**Pull request by gaagul**
Fixes #470 

-  Used Fuse.js to implement the fuzzy search logic for mentions.

### CODE REVIEW

**Review by AbhayVAshokan — CHANGES_REQUESTED** [review]
@gaagul _a while making the PR don't remove the default content provided by the PR template. You need to update the default content.

**Comment by gaagul** [comment]
@AbhayVAshokan Please Review.

File: lib/components/Editor/CustomExtensions/Mention/utils.js [hunk 1]
```diff
@@ -1,11 +1,23 @@
...
+    }
+
+    const options = {
+      keys: ["name"],
+    };
+    const fuse = new Fuse(mentions, options);
+    const results = fuse.search(query);
+
+    return pluck("item", results);
+  };
```
Comment (AbhayVAshokan) [inline lib/components/Editor/CustomExtensions/Mention/utils.js:22]: I used the fuzzy mentions feature for a seed data of 5000 users.

1. The behaviour for low threshold is what is expected. Add a `threshold` of `0.25`.
2. The current logic impacts the performance greatly. Slice the resulting array to 10 items or below.

@gaagul _a

Comment (leeoniya) [reply lib/components/Editor/CustomExtensions/Mention/utils.js:22]: Fuse is quite slow and produces poor matches. you might consider something else:

https://github.com/leeoniya/uFuzzy#a-biased-appraisal-of-similar-work

Comment (AbhayVAshokan) [reply lib/components/Editor/CustomExtensions/Mention/utils.js:22]: @leeoniya thank you for the suggestion. We'll definitely check out uFuzzy. For now we'll go forward with Fuse.js. We'll come back after creating a POC with uFuzzy and verifying the performance.

**Comment by gaagul** [comment]
@AbhayVAshokan Please review.

File: lib/components/Editor/CustomExtensions/Mention/utils.js [hunk 2]
```diff
@@ -1,11 +1,24 @@
...
 export const createMentionSuggestions = (users = []) => {
   const mentions = formatMentions(users);
 
-  return ({ query }) =>
-    mentions.filter(({ name }) =>
-      name.toLocaleLowerCase().includes(query.toLocaleLowerCase())
-    );
+  return ({ query }) => {
+    if (!query) {
+      return mentions;
```
Comment (AbhayVAshokan) [inline lib/components/Editor/CustomExtensions/Mention/utils.js:12]:
```suggestion
      return mentions.slice(0, 10);
```

@gaagul _a merge after making this minor change.

**Review by AbhayVAshokan — APPROVED** [review]

**Pull request merged**